common/djangoapps/util/memcache.py

```diff
@@ -8,15 +8,19 @@
 from urllib.parse import quote_plus
 
 from django.utils.encoding import smart_str
+from django.conf import settings
 
 
 def fasthash(string):
     """
     Hashes `string` into a string representation of a 128-bit digest.
     """
-    md4 = hashlib.new("md4")
-    md4.update(string.encode('utf-8'))
-    return md4.hexdigest()
+    if settings.FEATURES.get("ENABLE_BLAKE2B_HASH_FN", False):
+        hash_obj = hashlib.new("blake2b", digest_size=16)
+    else:
+        hash_obj = hashlib.new("md4")
+    hash_obj.update(string.encode('utf-8'))
+    return hash_obj.hexdigest()
```

Comment:
This is a take it or leave it one. blake2b constructor is [always present](https://docs.python.org/3/library/hashlib.html#hash-algorithms) so this can be simplified to:
```suggestion
    if settings.FEATURES.get("ENABLE_BLAKE2B_HASH_FN", False):
        return hashlib.blake2b(string.encode('utf-8'), digest_size=16).hexdigest()

    md4 = hashlib.new("md4")
    md4.update(string.encode('utf-8'))
    return md4.hexdigest()
```